File: c/convert/binaryToDecimal.c
#include <stdio.h>
#include <math.h>


double convertBinaryToDecimal(int *binary, int length) {
    double decimal = 0;
    double powerOfTwo = 1; 

    for (int i = length - 1; i >= 0; i--) {
        if (binary[i] == 1) {
            decimal += powerOfTwo;
        }
        powerOfTwo *= 2; 
    }

    return decimal;
}

int main() {
    int binary[] = {1, 1, 1, 1}; // 15
    int length = sizeof(binary) / sizeof(binary[0]);
    int decimal = 0;

    printf("Binário: ");

    for (int i = 0; i < length; i++) {
        printf("%d", binary[i]);
    }

    decimal = convertBinaryToDecimal(binary, length);

    printf("\nDecimal: %d\n", decimal);

    return 0;
}

File: java/src/main/java/com/marlonzl7/convert/BinaryToDecimal.java
package com.marlonzl7.convert;

public class BinaryToDecimal {
    public static void main(String[] args) {
        int[] binary = {1, 0, 0, 1};

        System.out.print("Binário: ");
        for (int i = 0; i < binary.length; i++) {
            System.out.printf("%d", binary[i]);
        }

        System.out.printf("\nDecimal: %d\n", convertBinaryToDecimal(binary));

    }

    public static int convertBinaryToDecimal(int[] binary) {
        int decimal = 0;

        for (int i = binary.length - 1; i >= 0; i--) {
            int j = binary.length - i - 1;
            decimal += (binary[i] == 1) ? (int) Math.pow(2, j) : 0;
        }

        return decimal;
    }
}
